### ISSUE TITLE
Empty username valid when registering

### ISSUE THREAD

**Issue by tqwewe** [strapi/strapi#8632]
### Describe the bug

When registering a new user from the API, it is valid to have an empty username, but only once. After registering an account with an empty username, new users cannot do the same as "Username is already taken".

### Steps to reproduce the behavior

1. Using the GraphQL API, create a new user:
```gql
mutation {
  register(input: {
    username: ""
    email: "someone@example.com"
    password: "password"
  }) {
    user {
      id
    }
  }
}
```
2. Change the email and run the mutation again.
3. Notice the first user registered successfully, and the next one failed.

### Expected behavior

No user should be able to register with empty usernames.

### Screenshots

N/A

### Code snippets

N/A

### System

- Node.js version: v14.4.0
- NPM version: v14.4.0
- Strapi version: 3.2.5
- Database: postgres
- Operating system: Mac OSX

### Additional context

I noticed this is a duplicate of issue https://github.com/strapi/strapi/issues/5496, but it seems like there was no clear resolution and it was just closed.

**Comment by derrickmehaffy** [comment]
Was able to confirm this on v3.3.3

**Comment by alexandrebodin** [comment]
This was fixed in v4 :) Going to close the issue
